tide-slog/src/lib.rs

```diff
@@ -1,50 +1,70 @@
-use slog::{info, o, Drain};
+#![feature(async_await)]
+#![warn(
+    nonstandard_style,
+    rust_2018_idioms,
+    future_incompatible,
+    missing_debug_implementations
+)]
+
+use slog::{info, trace, o, Drain};
 use slog_async;
 use slog_term;
 
 use futures::future::BoxFuture;
 use futures::prelude::*;
 
-use crate::{
+use tide_core::{
     middleware::{Middleware, Next},
     Context, Response,
 };
 
-/// Root logger for Tide. Wraps over logger provided by slog.SimpleLogger
+/// RequestLogger based on slog.SimpleLogger
 #[derive(Debug)]
-pub struct RootLogger {
+pub struct RequestLogger {
     // drain: dyn slog::Drain,
-    inner_logger: slog::Logger,
+    inner: slog::Logger,
 }
 
-impl RootLogger {
-    pub fn new() -> RootLogger {
-        let decorator = slog_term::TermDecorator::new().build();
-        let drain = slog_term::CompactFormat::new(decorator).build().fuse();
-        let drain = slog_async::Async::new(drain).build().fuse();
+impl RequestLogger {
+    pub fn new() -> Self {
+        Default::default()
+    }
 
-        let log = slog::Logger::root(drain, o!());
-        RootLogger { inner_logger: log }
+    pub fn with_logger(logger: slog::Logger) -> Self {
+        Self { inner: logger }
     }
 }
 
-impl Default for RootLogger {
+impl Default for RequestLogger {
```

Comment:
I don't think this should be provided at all, IMO if someone is going to the effort of using `slog` that's because they care strongly about their logging and will want full control over the root logger.

It also unconditionally introduces the `slog_term` and `slog_async` dependencies even if a user is going to use a different root logger.

Having a doc example show how to construct this together with `slog_term` etc. seems enough.

Reply:
Sure, I'm not even sure if `tide-slog` should be a part of tide repo, since it's a highly opinionated option. I just decided to make in a tad bit more configurable than it was previously extracting this out. 

(Hence, removed this from `tide::middleware`)

Reply:
Also, I personally don't use slog - now that it's out in it's own crate, think it'd be better with someone who actually uses it to pick this up and continue. /cc @bIgBV

Reply:
I originally added slog for the structured logging it provides. But given that the `log` crate is incorporating structured logging as well, I don't think we need to depend on slog again. We could look into providing support for `tokio-trace` once it stabilizes a bit.

Reply:
`slog`s main value add for something like a web server is _scoped_ logging. Being able to inject a request id onto the logger context at the start of the request and have that automatically emitted with every log line during that request is super useful.

I have an idea of a project I'm tempted to start writing with Tide, if I do get around to doing that I'll definitely be using `tide-slog` and would be happy to help with core maintainership of it.